src/fiskaltrust.Middleware.Demo/RestPos.cs

```diff
@@ -0,0 +1,210 @@
+﻿using fiskaltrust.ifPOS.v0;
+using fiskaltrust.ifPOS.v1;
+using fiskaltrust.Middleware.Demo.Helpers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using System.Xml.Serialization;
+
+namespace fiskaltrust.Middleware.Demo
+{
+    public class RestPos : ifPOS.v1.IPOS
+    {
+        private string _url;
+        private string _requestType;
+        public RestPos(string url)
+        {
+            _url = url.StartsWith("rest://") ? url.Replace("rest://", "http://") : url.Replace("xml://", "http://");
+            _requestType = url;
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginEcho(string message, AsyncCallback callback, object state)
+        {
+            throw new NotSupportedException();
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginJournal(long ftJournalType, long from, long to, AsyncCallback callback, object state)
+        {
+            throw new NotSupportedException();
+        }
+
+        IAsyncResult ifPOS.v0.IPOS.BeginSign(ifPOS.v0.ReceiptRequest data, AsyncCallback callback, object state)
+        {
+            throw new NotSupportedException();
+        }
+
+        string ifPOS.v0.IPOS.Echo(string message)
+        {       
+
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(_url);
+                
+                var data = new { message = message};
+                var jsonstring = JsonConvert.SerializeObject(data);
+                var jsonContent = new StringContent(jsonstring, Encoding.UTF8, "application/json");                                            
+
+                using (var response = client.PostAsync("v0/Echo", jsonContent).Result)
```

Comment:
Do you think it makes sense to wrap this functionality (httpcall with stringconent and deserialize) into a separate method to share this code?